### PR TITLE
Adding "jaxoplanet core from scratch" tutorial

### PR DESCRIPTION
**Pull request by dfm**
From docathon.

### CODE REVIEW

**Comment by soichiro-hattori** [comment]
Whoops sorry @dfm I wasn't trying to directly push to this upstream branch. I'll be more careful from now. 
But the `index.md` conflict is at least gone? 👍

**Comment by dfm** [comment]
@soichiro-hattori — No problem at all! Thanks for fixing that! I'll ask you to review this PR properly soon, but I need to do a little more writing on it first.

**Comment by dfm** [comment]
@soichiro-hattori — I think this is good to go. Can you give it a review? The history has gotten gnarly, but I don't think it'll be a problem since the merge queue will appropriately squash the commits!

**Comment by soichiro-hattori** [comment]
@dfm awesome! I'll take a look at this tomorrow and edit this comment with thoughts!

File: docs/tutorials/core-from-scratch.ipynb [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+    "We should note that the point here is not to pick on RedVel, and let's be clear that this is by no means a complete comparison!\n",
+    "Instead, we want to highlight that in some cases JIT-compiled JAX code can get similar performance to that provided by optimized low level code with Python bindings.\n",
+    "\n",
+    "Another benefit of this pure-JAX implementation is that it can be natively executed on hardware accelerators like a GPU or TPU.\n",
+    "For comparison, on an Nvidia V100 GPU, the above benchmark takes about `0.1 ms` (nearly 2 orders of magnitude faster than my CPU).\n",
+    "\n",
+    "### Supporting differentiation\n",
+    "\n",
+    "Another key feature of JAX (besides the fact that its JIT compiler produces very fast code) is that it supports automatic differentiation.\n",
+    "While it is already possible to differentiate the above code, that wouldn't produce the most efficient method.\n",
```
Comment (soichiro-hattori) [inline docs/tutorials/core-from-scratch.ipynb:224]: Maybe very briefly discuss why the built-in AD wouldn't be efficient? 
Even someone short like "...since tracing through the `kepler_solver_impl` to obtain the derivative is inefficient when we can explicitly write the derivative given below."

File: docs/tutorials/core-from-scratch.ipynb [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    "## Solving Kepler's equation\n",
+    "\n",
+    "One core piece of infrastructure provided by `jaxoplanet` is a function to solve Kepler's equation\n",
+    "\n",
+    "$$\n",
+    "M = E - e \\sin(E)\n",
+    "$$\n",
+    "\n",
+    "for the eccentric anomaly $E$ as a function of the eccentricity $e$ and mean anomaly $M$.\n",
+    "There is a lot of literature dedicated to solving this equation efficiently and robustly, and we won't get into all the details here, but there are a few points we should highlight:\n",
```
Comment (soichiro-hattori) [inline docs/tutorials/core-from-scratch.ipynb:29]: I don't have strong feelings but even some extremely short context as to why it's important to solve the Kepler equation might be nice? 

"There is a lot of literature dedicated to solving this equation efficiently and robustly _as it provides various geometric properties for the motion of a body orbiting a central object_,..."

**Review by soichiro-hattori — APPROVED** [review]

**Comment by soichiro-hattori** [comment]
Actually sorry @dfm there's one quick thing I wanted to check! Since we're using custom derivatives I was wondering if it might be worth showing the runtime of the traced derivatives to the custom one.

File: docs/tutorials/core-from-scratch.ipynb [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+    "ax.set_xlim(0, 2 * jnp.pi);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "15",
+   "metadata": {},
+   "source": [
+    "Finally, we can test the performance of this operation to see that the computational cost is still dominated by the solve.\n",
+    "In other words, the total cost of the of computing these derivatives is only fractionally (about 1% on my laptop) higher than just solving the equation itself."
```
Comment (soichiro-hattori) [inline docs/tutorials/core-from-scratch.ipynb:324]: Perhaps add "In comparison, the runtime of the default traced gradient computation is higher by a factor of four."

**Comment by soichiro-hattori** [comment]
@dfm: Okay that's all I had in mind!

File: docs/tutorials/core-from-scratch.ipynb [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "ecc = 0.5\n",
+    "true_ecc_anom = jnp.linspace(0, 2 * jnp.pi, 50_000)[:-1]\n",
+    "mean_anom = true_ecc_anom - ecc * jnp.sin(true_ecc_anom)\n",
+    "\n",
+    "calc_acc_anom = kepler_solver_impl(mean_anom, ecc)\n",
+    "\n",
+    "plt.plot(true_ecc_anom, jnp.abs(calc_acc_anom - true_ecc_anom), \"k\")\n",
```
[the remark this answers is not in the document]
Comment (dfm) [reply docs/tutorials/core-from-scratch.ipynb:174]: Fixed!

File: docs/tutorials/core-from-scratch.ipynb (hunk 1 above)
Comment (dfm) [reply docs/tutorials/core-from-scratch.ipynb:224]: Added!

File: docs/tutorials/core-from-scratch.ipynb (hunk 2 above)
Comment (dfm) [reply docs/tutorials/core-from-scratch.ipynb:29]: Done.

File: docs/tutorials/core-from-scratch.ipynb (hunk 3 above)
Comment (dfm) [reply docs/tutorials/core-from-scratch.ipynb:324]: Added.

**Comment by dfm** [comment]
Thanks @soichiro-hattori!!

I've incorporated all your changes and I'm going to merge.
